crates/broker/src/order_monitor.rs

```diff
@@ -211,42 +299,411 @@ where
+                    .await
+                    .context("Failed to set order status to skipped")?;
+            } else if let Some((locker, _)) =
+                self.db.get_request_locked(U256::from(order.request.id)).await?
+            {
+                let our_address = self.provider.default_signer_address().to_string().to_lowercase();
+                let locker_address = locker.to_lowercase();
+                if locker_address != our_address {
+                    tracing::info!("Request {:x} was scheduled to be locked by us ({}), but is already locked by another prover ({}). Skipping.", order.request.id, our_address, locker_address);
+                    self.db
+                        .set_order_status(&order.id(), OrderStatus::Skipped)
+                        .await
+                        .context("Failed to set order status to skipped")?;
+                } else {
+                    // Edge case where we locked the order, but due to some reason was not moved to proving state. Should not happen.
+                    tracing::info!("Request {:x} was scheduled to be locked by us, but is already locked by us. Proceeding to prove.", order.request.id);
+                    candidate_orders.push(order);
+                }
+            } else {
+                candidate_orders.push(order);
+            }
+        }
+
+        if candidate_orders.is_empty() {
+            tracing::trace!(
+                "No orders to lock and/or prove as of block timestamp {}",
+                current_block_timestamp
+            );
+            return Ok(Vec::new());
+        }
+
+        let mut final_orders: Vec<Order> = Vec::new();
+        for order in candidate_orders {
+            let now = now_timestamp();
+            if order.request.expires_at() < current_block_timestamp {
+                tracing::debug!("Request {:x} has now expired. Skipping.", order.request.id);
+                self.db
+                    .set_order_status(&order.id(), OrderStatus::Skipped)
+                    .await
+                    .context("Failed to set order status to skipped")?;
+            } else if order.request.expires_at() - now < min_deadline {
+                tracing::debug!("Request {:x} deadline at {} is less than the minimum deadline {} seconds required to prove an order. Skipping.", order.request.id, order.request.expires_at(), min_deadline);
+                self.db
+                    .set_order_status(&order.id(), OrderStatus::Skipped)
+                    .await
+                    .context("Failed to set order status to skipped")?;
+            } else {
+                final_orders.push(order);
+            }
+        }
+
+        tracing::info!(
+            "After filtering invalid orders, found total of {} valid orders to proceed to locking and/or proving", 
+            final_orders.len()
+        );
+        tracing::debug!(
+            "Final orders ready for locking and/or proving after filtering: {final_orders:?}"
```

Comment:
sorry I mixed up logs from https://github.com/boundless-xyz/boundless/pull/542/files/574d87e616bbb2be90aacc27194db23cc7fdabfa#r2066818924 this was the one I meant to point to. Perhaps if info can be useful it's made as a trace log for our debugging purposes? Debug is on by default with our default broker setup right now

Reply:
Oh, so I implemented a custom debug for this that keeps it short: https://github.com/boundless-xyz/boundless/pull/542/files#diff-99ebc229b742243f5d295e008af90162c013ee6455ac3d0c49f3f86ef9c2295eR84-R95

I've found it quite useful to have on debug mode, but happy to move to trace if you feel strongly